apps/ponder-interop/src/api/index.ts

```diff
@@ -233,6 +271,81 @@ function getPendingClaimsQuery(relayers: Address[]) {
     .orderBy(desc(schema.gasTankRelayedMessageReceipts.relayedAt))
 }
 
+/**
+ * Fetches pending claims for a given gas provider
+ * @param gasProvider - The gas provider address
+ * @param relayers - The relayers to fetch pending claims for
+ * @param chainId - The chain ID to fetch pending claims for
+ * @returns The pending claims
+ */
+function getPendingClaimsForGasProviderQuery(
+  gasProvider: Address,
+  relayers: Address[],
```

Comment:
❓ @tremarkley can you remind me the relationship between the gasProvider and the relayer? I thought the relayer also provided the gas

Reply:
The relayer is the sender of the relay tx and therefore pays the gas up front. The `gasProvider` holds a balance on the `GasTank` contract and that balance is used to refund the relayer for the gas that it paid when relaying the tx ([here](https://github.com/ethereum-optimism/interop-lib/blob/3d7391a4433616cc618e5f4ffed294d39b09177c/src/experiment/GasTank.sol#L174) in teh `claim` function on the `GasTank` is where the relayer gets refunded). In addition to relaying messages, the relayer is also responsible for calling the `claim` function on the `GasTank` in order to receive its refund. This query just returns the amount of outstanding claims that could be claimed against a given `gasProvider`

Reply:
👍 ohh makes sense, i was thinking of "providing gas" like "who paid gas to relay the message", but it means "who paid into the tank to have a message relayed"